eldritch/events.py

```diff
@@ -1938,6 +1945,41 @@ def start_str(self):
   def finish_str(self):
     return f"{self.returned} monsters returned to the cup"
 
+class CloseLocation(Event):
+
+  def __init__(self, location_name, for_turns=float('inf'), evict=True):
+    self.location_name = location_name
+    self.for_turns = for_turns
+    self.evict = evict
+    self.resolved = False
+
+  def resolve(self, state):
+    until = state.turn_number + self.for_turns + 1
+    place = state.places[self.location_name]
+    place.closed_until = until
+    chars_in_place = [char for char in state.characters if char.place == place]
+    monsters_in_place = [mon for mon in state.monsters if mon.place == place]
+
+    if place.closed and self.evict:
+      to_place = next(iter(place.connections))
+      for char in chars_in_place:
+        state.event_stack.append(ForceMovement(char, to_place.name))
+        return False
```

Comment:
This is...strange. I guess it works in that you exit the for loop after the first iteration, so you only append one event at a time. Appending multiple events to the loop at once is disallowed, but it looks like you're not doing that.

I would feel better if this were a Sequence of ForceMovements, but I can't actually justify that with anything.

Reply:
Agree, changed.